src/python/pants/option/config.py

```diff
@@ -135,46 +134,30 @@ def get(self, section, option) -> str | None:
         If the specified section does not exist or is missing a definition for the option, the value
         is looked up in the DEFAULT section.  If there is still no definition found, returns None.
         """
-        if not self.has_option(section, option):
-            return None
-        return self.get_value(section, option)
-
-    @abstractmethod
-    def configs(self) -> Sequence[_SingleFileConfig]:
-        """Returns the underlying single-file configs represented by this object."""
+        for vals in reversed(self.values):
+            val = vals.get_value(section, option)
+            if val is not None:
+                return val
+        return None
 
-    @abstractmethod
     def sources(self) -> list[str]:
         """Returns the sources of this config as a list of filenames."""
+        return [vals.path for vals in self.values]
 
-    @abstractmethod
-    def sections(self) -> list[str]:
-        """Returns the sections in this config (not including DEFAULT)."""
-
-    @abstractmethod
-    def has_section(self, section: str) -> bool:
-        """Returns whether this config has the section."""
-
-    @abstractmethod
-    def has_option(self, section: str, option: str) -> bool:
-        """Returns whether this config specified a value for the option."""
-
-    @abstractmethod
-    def get_value(self, section: str, option: str) -> str:
-        """Returns the value of the option in this config as a string.
-
-        Raises NoSectionError if the section doesn't exist. Raises NoOptionError if the option
-        doesn't exist in the section.
-        """
-
-    @abstractmethod
     def get_source_for_option(self, section: str, option: str) -> str | None:
-        """Returns the path to the source file the given option was defined in.
+        """Returns the path(s) to the source file(s) the given option was defined in.
+
+        If an option was defined in multiple files, returns a comma-separated list.
```

Comment:
How about returning a `list[str]`, then callers can combine how they want? That seems more natural. And you can get rid of the `None` by using an empty list.

Reply:
Oh yeah, and the caller calls relpath on the result anyway, so this is actually necessary. Done.

Reply:
Although it probably makes more sense to call relpath in the method and not the caller, so will do that.